scripts/error_report_utils.sh

```diff
@@ -0,0 +1,43 @@
+# Some utils functions for scripts to help report errors correctly
+
+# counting for errors
+# ERRORS is the number of errors
+ERRORS=0
+# FAILED_COMMANDS is a list of commands that failed
+FAILED_COMMANDS=""
+
+# reset the error count
+init_error_report () {
+    ERRORS=0
+    FAILED_COMMANDS=""
+}
+
+# account for an error
+# modifies the ERRORS and FAILED_COMMANDS variables
+function report_error () {
+    return_code=$1
+    script_name=$2
+    if [ "$return_code" -ne 0 ];
+    then
+        >&2 echo "ERROR: $script_name not executed successfully - return value: $return_code"
+        ERRORS=`expr $ERRORS + 1`
+        FAILED_COMMANDS="${FAILED_COMMANDS}$script_name
+    "
+    fi
+}
+
+# If there were commands that resulted in errors,
+# echo the list of commands so that it is included in the
+# failure e-mail sent to root
+function report_failed_commands () {
+    if [ $ERRORS -gt 0 ];
+    then
+        >&2 echo "ERROR: $ERRORS ERROR(S) DURING EXECUTION OF gen_fields_daily.sh"
```

Comment:
just a small detail, this script is also now used by other scripts than gen_feeds_daily.sh like generate_madenearme_pages.sh